cirq-core/cirq/testing/circuit_compare.py

```diff
@@ -237,7 +237,11 @@ def assert_has_diagram(
     # pylint: enable=unused-variable
     actual_diagram = actual.to_text_diagram(**kwargs).lstrip("\n").rstrip()
     desired_diagram = desired.lstrip("\n").rstrip()
-    assert actual_diagram == desired_diagram, (
+
+    def remove_subcircuit_labels(s: str):
+        return ''.join(line for line in s.splitlines() if 'Circuit_0x' not in line)
+
+    assert remove_subcircuit_labels(actual_diagram) == remove_subcircuit_labels(desired_diagram), (
```

Comment:
[No change required] I think there's an open issue to remove subcircuit labeling, so we should be able to remove this step at some point.